include/dlaf/eigensolver/bt_band_to_tridiag/mc.h

```diff
@@ -135,8 +138,131 @@ auto updateE(pika::threads::thread_priority priority, WSender&& tile_w, W2Sender
       pika::execution::experimental::start_detached();
 }
 
+struct Helper {
+  Helper(const SizeType b, const SizeType nrefls, matrix::Distribution dist,
+         const GlobalElementIndex offset)
+      : nrefls_(nrefls), input_spec_{dist.tileElementIndex(offset),
+                                     {std::min(b, dist.size().rows() - offset.row()),
+                                      std::min(b, dist.size().cols() - offset.col())}} {
+    const TileElementIndex& sub_offset = input_spec_.origin;
+
+    // Note:
+    // Next logic is about detecting the available application space for reflectors, which once
+    // extracted they expands to a matrix with 2 * b - 1 height.
+    //
+    // There are two main scenarios:
+    // - reflectors involves rows of a single tile;
+    // - reflectors involves rows across two different tiles.
+    //
+    // In both scenarios, it may happen that all reflectors cannot be fully applied to to matrix
+    // size constraint.
+
+    const SizeType fullsize = 2 * b - 1;
+    const SizeType mb = dist.blockSize().rows();
+    const SizeType rows_below = dist.size().rows() - offset.row();
+
+    // Note:
+    // In general, keep in mind that to sub_offset.row() is applied a + 1, to take into account the
+    // offset about how reflectors are actually applied.
+    //
+    // e.g. b = 4
+    // reflectors   matrix
+    //              X X X X
+    // 1 0 0 0      X X X X
+    // a 1 0 0      X X X X
+    // a b 1 0      X X X X
+    //              -------
+    // a b c 1      Y Y Y Y
+    // 0 b c d      Y Y Y Y
+    // 0 0 c d      Y Y Y Y
+    // 0 0 0 d      Y Y Y Y
+    //
+    // From the drawing above, it is possible to see the dashed tile separation between X and Y,
+    // and how the reflectors on the left are going to be applied. In particular, the first row of
+    // the upper tile is not affected.
+
+    acrossTiles_ = [&]() {
+      // Note:
+      // A single tile is involved if:
+      // - it is the last row tile, so by construction reflectors will be applied to a single tile;
+      // - applying reflectors falls entirely withing a single tile
+      const SizeType i_tile = dist.globalTileFromGlobalElement<Coord::Row>(offset.row());
+      const bool is_last_row_tile = i_tile == dist.nrTiles().rows() - 1;
+      return !(is_last_row_tile || sub_offset.row() + 1 + fullsize <= mb);
+    }();
```

Comment:
Quite a lot of setup seems do be done in the constructor of this struct. Is it possible to refactor this into separate functions with names?

Reply:
IMHO it is a big comment chunk, but in the end it does two (linked) things:
- determine if there is just 1 or 2 parts (`acrossTiles_`)
- create descriptors for existing part(s)

Do you think it would be worth splitting them?

Reply:
I think it's not a blocker. Perhaps we can do it in the future.